### PR TITLE
Add architectural decision record for the consignment reference work

### PR DESCRIPTION
**Pull request by Lholbourn**
This describes the options/tools considered for both the encoding of the base25 section of the consignment reference and the incremental ID which will be used to create the base25 reference.

### CODE REVIEW

File: architecture-decision-records/0011-base25-encoding-for-consignment-reference.md
Comment (suzannehamilton) [inline architecture-decision-records/0011-base25-encoding-for-consignment-reference.md:18]: Could you mention that we're not trying to match specific IDs used in those other systems? For example, we're not trying to generate the IDs for DRI or the Omega catalogue - they will continue to use their own ID generation.

File: architecture-decision-records/0011-base25-encoding-for-consignment-reference.md
Comment (suzannehamilton) [inline architecture-decision-records/0011-base25-encoding-for-consignment-reference.md:17]: Also, the reason to use a system like this rather than just a number (e.g. TDR-2020-12345) is that the encoded part is much shorter than the equivalent integer once the numbers grow large, so it's easier for users to read.

File: architecture-decision-records/0011-base25-encoding-for-consignment-reference.md
Comment (suzannehamilton) [inline architecture-decision-records/0011-base25-encoding-for-consignment-reference.md:13]: It's probably worth mentioning that the reason we use the consignment creation year is that a transfer might takes days or weeks to finish, so we don't know the completion year when we generate the reference. We have to generate the reference at the start of the process so we can display it on pages like the transfer summary page and the transfer history page.

File: architecture-decision-records/0011-base25-encoding-for-consignment-reference.md
Comment (suzannehamilton) [inline architecture-decision-records/0011-base25-encoding-for-consignment-reference.md:9]: I'd include an example as well, which will help clear up any ambiguity.

File: architecture-decision-records/0011-base25-encoding-for-consignment-reference.md
Comment (suzannehamilton) [inline architecture-decision-records/0011-base25-encoding-for-consignment-reference.md:15]: Is this part in upper case or lower case? (Unless it's mixed, but I don't think that's how the encoding algorithm works?)

File: architecture-decision-records/0011-base25-encoding-for-consignment-reference.md
Comment (suzannehamilton) [inline architecture-decision-records/0011-base25-encoding-for-consignment-reference.md:7]: It would also be good to add the context that the reference is intended to be a human-readable reference. We already use UUIDs to identify consignments that go through TDR, but UUIDs are hard for users to recognise or type/read out.

I think they current plan is that we'll export both the TDR UUID and the consignment reference to the long-term preservation system, so we can use either ID to look up a consignment in the future.

File: architecture-decision-records/0011-base25-encoding-for-consignment-reference.md
Comment (suzannehamilton) [inline architecture-decision-records/0011-base25-encoding-for-consignment-reference.md:84]: Can you reiterate in this section that the format will be `TDR-<year>-<base25-ID>`? Just so it's clear that this is part of the decision we're recording here.

File: architecture-decision-records/0011-base25-encoding-for-consignment-reference.md
Comment (suzannehamilton) [inline architecture-decision-records/0011-base25-encoding-for-consignment-reference.md:5]: Could you also mention the things that we considered including in the reference, and why we decided against them? It's the kind of thing we'll forget in a year's time.

I think we talked about:

* Series ID: this was proposed because a consignment currently belongs to a single series, so it might be a useful reference to users. We decided against it because the long-term plan is to remove the concept of series from TDR.
* Transferring body code or name: proposed for the same reason as a series. I'm actually not sure why we decided against this one. Was it because departments can change (e.g. name changes, merges and splits) so it's not as stable an identifier as it would seem?

**Review by suzannehamilton — COMMENTED** [review]
This looks really good, thanks! 🎉 

I've added quite a lot of comments - it's just because I don't want us to forget anything we've discussed. 😅

File: architecture-decision-records/0011-base25-encoding-for-consignment-reference.md
Comment (techncl) [inline architecture-decision-records/0011-base25-encoding-for-consignment-reference.md:26]: Just noticed a typo "assocciated" rather than "associated"

File: architecture-decision-records/0011-base25-encoding-for-consignment-reference.md
Comment (Lholbourn) [reply architecture-decision-records/0011-base25-encoding-for-consignment-reference.md:18]: GCR is Generated Catalogoue References - the only place I see it discussed is the DRI documentation. I'll add an explanation for that!

File: architecture-decision-records/0011-base25-encoding-for-consignment-reference.md
Comment (Lholbourn) [reply architecture-decision-records/0011-base25-encoding-for-consignment-reference.md:15]: It's uppercase, I'll change that to reflect it! :+1:

File: architecture-decision-records/0011-base25-encoding-for-consignment-reference.md
Comment (suzannehamilton) [inline architecture-decision-records/0011-base25-encoding-for-consignment-reference.md:16]: I think it's worth clarifying that the users will see consignment references on TDR pages, but the URLs will continue to use the UUIDs. (Just in case someone in future notices the URLs and is confused because they are technically user-visible.)

Comment (Lholbourn) [reply architecture-decision-records/0011-base25-encoding-for-consignment-reference.md:16]: Ooh good point! I'll change that :+1:

**Comment by suzannehamilton** [comment]
This looks good to me, thanks! I won't approve it yet, in case anyone else has comments.

**Review by suzannehamilton — APPROVED** [review]
I'll approve this now so you can merge it when you're happy that you've got all the feedback you were expecting.

**Pull request merged**
